cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -166,14 +166,14 @@ public void init(ProgramController.State state, @Nullable Throwable cause) {
         public void completed() {
           LOG.debug("Program {} {} {} completed successfully.", id.getNamespaceId(), id.getApplicationId(), id.getId());
           store.setStop(id, runId, TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()),
-                        ProgramController.State.COMPLETED.getRunStatus());
+                        ProgramController.State.COMPLETED.getRunStatus(), null);
```

Comment:
Better to have another `store.setStop` method that doesn't need the failure cause to pass it and default it to `null`.